templates/ovncontroller/bin/readiness.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+# Function to handle errors and exit with a message
+error_exit() {
+    echo "$1"
+    exit 1
+}
+
+# Check if ovn-controller is connected to the OVN SB database
+check_ovn_controller_connection() {
+    output=$(ovn-appctl -t ovn-controller connection-status)
+    if [ $? -ne 0 ]; then
+        error_exit "ERROR - Failed to get connection status from ovn-controller"
+    fi
+    if [ "$output" != "connected" ]; then
+        error_exit "ERROR - OVN-Controller is not connected to OVN SB database"
+    fi
+}
+
+# Check if ovsdb-server is running and list its databases
+check_ovsdb_server() {
+    ovsdbserverpidfile=$(find /var/run/openvswitch -name "ovsdb-server.pid")
+    if [ -z "$ovsdbserverpidfile" ]; then
+        error_exit "ERROR - Failed to find ovsdb-server PID file"
+    fi
+
+    pid=$(cat "$ovsdbserverpidfile")
+    if [ -z "$pid" ]; then
+        error_exit "ERROR - Failed to read PID from ovsdb-server PID file"
+    fi
+
+    output=$(ovn-appctl -t /var/run/openvswitch/ovsdb-server."$pid".ctl ovsdb-server/list-dbs)
+    if [ $? -ne 0 ]; then
+        error_exit "ERROR - Failed to communicate with ovsdb-server"
+    fi
+
+    if [ -z "$output" ]; then
```

Comment:
If you go with it, at least confirm that the expected database name is in the output? (Not sure we should do the check for any of these though - this strikes me as an implementation detail that the config is in a local ovsdb-server.)

Reply:
By previous comment reason, removing last checks over ovsdb-server and ovs-vswitchd.